### PR TITLE
Various fixes for the Subsonic provider

### PR DESCRIPTION
**Pull request by khers**
Update py-opensonic to 5.2.0 for the updated song parsing (now includes disc_number).

With the new disc_number field, pass this information on to MA for better track sorting.

Finally, we have yet another attempt to fix the Navidrome multi-artist disc issue. The original hack did not include the calculated item id and overloaded all artists that needed it with UNKNOWN_ARTIST instead causing odd artists to be listed.

### CODE REVIEW

File: music_assistant/server/providers/opensubsonic/sonic_provider.py [hunk 1]
```diff
@@ -100,13 +100,15 @@ async def handle_async_init(self) -> None:
             success = await self._run_async(self._conn.ping)
             if not success:
                 msg = (
-                    f"Failed to connect to {self.config.get_value(CONF_BASE_URL)}, "
+                    "Failed to connect to "
+                    f"{self.config.get_value(CONF_BASE_URL)}, "
                     "check your settings."
                 )
                 raise LoginFailed(msg)
```
Comment (marcelveldt) [inline music_assistant/server/providers/opensubsonic/sonic_provider.py]: why not raise AuthError or CredentialError here and catch it in one sweep below ?

File: music_assistant/server/providers/opensubsonic/sonic_provider.py [hunk 2]
```diff
@@ -369,7 +372,7 @@ def _parse_track(self, sonic_song: SonicSong) -> Track:
             # We are setting disc number to 0 because the standard for what is part of
             # a Open Subsonic Song is not yet set and the implementations I have checked
             # do not contain this field. We should revisit this when the spec is finished
-            disc_number=0,
+            disc_number=sonic_song.disc_number if sonic_song.disc_number is not None else 0,
```
Comment (marcelveldt) [inline music_assistant/server/providers/opensubsonic/sonic_provider.py]:
```suggestion
            disc_number=sonic_song.disc_number or 0,
```

File: music_assistant/server/providers/opensubsonic/sonic_provider.py [hunk 3]
```diff
@@ -411,21 +414,23 @@ def _parse_track(self, sonic_song: SonicSong) -> Track:
                 # because it will not have an entry in the artists table so the best we can do it
                 # add a 'fake' id with the proper artist name and have get_artist() check for this
                 # id and handle it locally.
+                fake = f"{NAVI_VARIOUS_PREFIX}{sonic_song.artist}"
```
Comment (marcelveldt) [inline music_assistant/server/providers/opensubsonic/sonic_provider.py]: make fake_id would be a better name for this variable ?

File: music_assistant/server/providers/opensubsonic/sonic_provider.py (hunk 1 above)
Comment (khers) [reply music_assistant/server/providers/opensubsonic/sonic_provider.py]: It felt silly to raise the exception to be caught two lines later and repackaged into the LoginFailed anways, but will do

**Review by marcelveldt — APPROVED** [review]

**Pull request merged**